Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationResultDispatchingValidator.cs

```diff
@@ -36,16 +36,71 @@ public void DispatchValidationFailures (IBusinessObjectValidationResult validati
       ArgumentUtility.CheckNotNull("validationResult", validationResult);
 
       var bocListControl = GetControlToValidate();
+      var dispatchToValidationFailureRepository = bocListControl
+          .GetColumnDefinitions()
+          .Any(cd => cd is BocValidationErrorIndicatorColumnDefinition);
 
+      if (dispatchToValidationFailureRepository)
+        DispatchToValidationFailureRepository(validationResult, bocListControl);
+      else
+        DispatchToValidationResultDispatchers(validationResult, bocListControl);
+    }
+
+    private void DispatchToValidationResultDispatchers (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
       var validatorsMatchingToControls = EnumerableUtility.SelectRecursiveDepthFirst(
               bocListControl as Control,
-              child => child.Controls.Cast<Control>().Where(item => !(item is INamingContainer)))
+              child => child.Controls.Cast<Control>().Where(item => item is not INamingContainer))
           .OfType<IBusinessObjectBoundEditableWebControlValidationResultDispatcher>();
 
       foreach (var validator in validatorsMatchingToControls)
         validator.DispatchValidationFailures(validationResult);
     }
 
+    private void DispatchToValidationFailureRepository (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
+      var validationFailureRepository = bocListControl.ValidationFailureRepository;
+      validationFailureRepository.ClearAllValidationFailures();
+
+      var rowObjects = bocListControl.Value?.Cast<IBusinessObject>();
+
+      if (rowObjects != null)
+      {
+        var columnsAndMatchers = bocListControl
+            .GetColumnDefinitions()
+            .OfType<IBocColumnDefinitionWithValidationSupport>()
+            .Select(c => new { Column = (BocColumnDefinition)c, Matcher = c.GetValidationFailureMatcher() })
+            .ToArray();
+
+        foreach (var rowObject in rowObjects)
+        {
+          foreach (var columnAndMatcher in columnsAndMatchers)
+          {
+            var matchedFailures = columnAndMatcher.Matcher.GetMatchingValidationFailures(rowObject, validationResult);
+
+            if (matchedFailures.Count > 0)
+              validationFailureRepository.AddValidationFailuresForDataCell(rowObject, columnAndMatcher.Column, matchedFailures);
+          }
+
+          var otherRowFailures = validationResult.GetValidationFailuresForBusinessObject(rowObject);
+          if (otherRowFailures.Count > 0)
+            validationFailureRepository.AddValidationFailuresForDataRow(rowObject, otherRowFailures);
```

Comment:
Formitting consistency: I think we should remove the blank line in line 80. It looks weirder if we have a blank line in line 80 and line 86 than without blank lines.